docs/source/getting-started/first-feature.rst

```diff
@@ -5,124 +5,178 @@ Your First Feature
 
-Message Edit Component
-----------------------
+MessageEdit Component
+---------------------
+
+Next we need the ``MessageEdit`` component to compose and send messages to selected friends.
+Again we show the entire component here; you should copy this into a new file in ``ui/src/components``.
 
-In addition to the feed component, we need a component for composing messages and sending them using the appropriate choice on the ``User`` contract.
+.. TODO Include file in template with placeholder for component logic.
 
 .. literalinclude:: code/ui-after/MessageEdit.tsx
 
-In this component we use React hooks to track the current choice of message receiver and content.
-The ``exerciseSendMessage`` hook gives us a function to exercise the appropriate choice on our ``User``.
-We wrap this in the ``sendMessage`` function to report potential errors to the user, and then the ``submitMessage`` function, called when the "Send" button is clicked.
-The ``isSubmitting`` state is used to ensure that message requests are processed one at a time.
-The result of each send is a new ``Message`` contract created on the ledger.
+You will first notice a ``Props`` type near the top of the file with a single ``friends`` field.
+A *prop* in React is an input to a component; in this case a list of users from which to select the message receiver.
+The prop will be passed down from the ``MainView`` component, reusing the work required to query users from the ledger.
+You can see this ``friends`` field bound at the start of the ``MessageEdit`` component.
+
+We use the React ``useState`` hook to get and set the current choices of message ``receiver`` and ``content``.
+The DAML-specific ``useExerciseByKey`` hook gives us a function to both look up a ``User`` contract and exercise the ``SendMessage`` choice on it.
+The call to ``exerciseSendMessage`` in ``sendMessage`` looks up the ``User`` contract with the receiver's username and exercises ``SendMessage`` with the appropriate arguments.
+The ``sendMessage`` wrapper reports potential errors to the user, and ``submitMessage`` additionally uses the ``isSubmitting`` state to ensure message requests are processed one at a time.
+The result of a successful call to ``submitMessage`` is a new ``Message`` contract created on the ledger.
+
+The return value of this component is the React ``Form`` element.
+This contains a dropdown menu to select a receiver from the ``friends``, a text field for the message content, and a *Send* button which triggers ``submitMessage``.
 
-View Component
---------------------
+There is again an important point here, in this case about how *authorization* is enforced.
+Due to the logic of the ``SendMessage`` choice, it is impossible to send a message to a user who has not added you as a friend (even if you could somehow access their ``User`` contract).
+The assertion that ``sender `elem` friends`` in ``SendMessage`` ensures this: no mistake or malice by the UI programmer could breach this.
```

Comment:
Let's not use the infix-with-backticks syntax in out example code. I don't think this language feature is very beginner friendly and doesn't add any value here.

Reply:
Agreed, I'll change it.

Reply:
Oh... DLint gives a warning if you use prefix `elem`. That might be worse.

Reply:
> Oh... DLint gives a warning if you use prefix `elem`. That might be worse.

Disable the warning? I agree with @hurryabit that infix syntax only adds one more thing that people have to learn so the warning just doesn’t make sense.

Reply:
Yeah, I can add a dlint.yaml file which can hopefully be easily ignored.

Reply:
I was more thinking of disabling it everywhere by default. IMHO if our default config does not work for the quickstart guide we should change the default config not overwrite it for quickstart.